gluten-substrait/src/main/scala/org/apache/gluten/substrait/rel/RelBuilderUtil.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gluten.substrait.rel
+
+import org.apache.gluten.backendsapi.BackendsApiManager
+import org.apache.gluten.expression.{ConverterUtils, ExpressionConverter}
+import org.apache.gluten.substrait.`type`.TypeBuilder
+import org.apache.gluten.substrait.SubstraitContext
+import org.apache.gluten.substrait.expression.ExpressionNode
+import org.apache.gluten.substrait.extensions.{AdvancedExtensionNode, ExtensionBuilder}
+
+import org.apache.spark.sql.catalyst.expressions.{Attribute, Expression}
+
+import java.util.{List => JList}
+
+import scala.collection.JavaConverters._
+
+object RelBuilderUtil {
```

Comment:
Any consideration not adding the APIs to `RelBuilder` directly?

Reply:
RelBuilder is a java class, add it in requires additional type conversion, such as scala `Seq`.

Reply:
There is also Java classes like `JList[ExpressionNode]` in this file. Can we just put them to RelBuilder? Then make caller call "asJava" when passing the  Scala collections which should be trivial. Otherwise developer will have to know 2 different helper classes for similar purposes.

Reply:
Mainly it seems inappropriate to call replaceWithExpressionTransformer in RelBuild. Here I extracted part of the logic, stay up replaceWithExpressionTransformer, CI failure unrelated, please take a look again, thank you.